anime_downloader/commands/ezdl.py

```diff
@@ -0,0 +1,194 @@
+)
+@click.option(
+    '--choice', '-c',type=int,
+    help='Choice to start downloading given anime number '
+)
+@click.option(
+    '--ratio', '-r',type=int,
+    help='Ratio used for the auto select in search. 100 means it only auto selects on complete matches. 0 auto selects regardless of how similar the result is.',
+    default=50
+)
+@click.option(
+    '--url', '-u', type=bool, is_flag=True,
+    help="If flag is set, prints the stream url instead of downloading")
+@click.option("--skip-fillers", is_flag=True, help="Skip downloading of fillers.")
+
+@click.pass_context
+def command(ctx, anime_url, episode_range, player, quality,
+            force_download, file_format, provider,
+            disable_ssl, choice, skip_fillers, ratio, url):
+    
+    # Borrows some config from the original dl command.
+    fallback_qualities = Config['dl']['fallback_qualities']
+    download_dir = Config['dl']['download_dir']
+    quality = Config['dl']['quality'] if not quality else quality
+    url = Config['dl']['url'] if not url else url
+    external_downloader = Config['dl']['external_downloader']
+    skip_download = Config['dl']['skip_download']
+    chunk_size = Config['dl']['chunk_size']
+    speed_limit = Config['dl']['speed_limit']
+
+    fallback_providers = Config['ezdl']['fallback_providers']
+
+    query = anime_url[:]
+    util.print_info(__version__)
+
+    fallback_providers.insert(0, provider)
+    # Eliminates duplicates while keeping order
+    providers = sorted(set(fallback_providers),key=fallback_providers.index)  
+    # TODO: flag for fallback providers
+
+    info = animeinfo.search_anilist(query)
+    episode_count = info.episodes - 1
+    # Interprets the episode range for use in a for loop.
+    # 1:3 -> for _episode in range(1, 4):
+    episode_range = util.parse_episode_range(episode_count, episode_range)
+    episode_range_split = episode_range.split(':')
+
+    # Stores the choices for each provider, to prevent re-prompting search.
+    choice_dict = {}
+
+    for episode_range in range(int(episode_range_split[0]), int(episode_range_split[-1])+1):
+        # Exits if all providers are skipped.
+        if [choice_dict[i] for i in choice_dict] == [0]*len(providers):
+            logger.info('All providers skipped, exiting')
+            exit()
+
+        for provider in providers:
+            if not get_anime_class(provider):
+                logger.info('"{}" is an invalid provider'.format(provider))
+                continue
+
+            logger.debug('Current provider: {}'.format(provider))
+            # TODO: Replace by factory
+            cls = get_anime_class(anime_url)
+
+            # To make the downloads use the correct name if URL:s are used.
+            real_provider = cls.sitename if cls else provider
+            # This will allow for animeinfo metadata in filename and one filename for multiple providers.
+            rep_dict = {
+                'animeinfo_anime_title': util.slugify(info.title),
```

Comment:
for readability I would prefer that `utils.slugify(info.title)` and variants were initialised outside of the dictionary with an informative variable name. The comment doesn't cut it for ease of initial reading